pages/samvera/developer_resources/iiif/hyrax_2.1/troubleshooting_riiif.md

```diff
@@ -0,0 +1,94 @@
+---
+title: "Troubleshooting RIIIF"
+keywords: IIIF
+categories: IIIF
+permalink: troubleshooting_riiif.html
+folder: samvera/how-to/iiif/hyrax_2.1/troubleshooting_riiif.md
+sidebar: home_sidebar
+tags: [development_resources]
+a-z: ['Troubleshooting RIIIF']
+version:
+  label: 'Hyrax v2.1.0 (also applies to Hyrax v2.0.x)'
+---
+
+# Troubleshooting RIIIF
+
+[RIIIF](https://github.com/curationexperts/riiif) is a gem for installing a simple [IIIF](http://iiif.io/) server into your rails application. Hyrax 2.1 ships with it, but it can be used in any rails application. This guide is for troubleshooting your installation if it isn't working as expected.
+
+This guide assumes you have installed RIIIF either using the instructions in [the RIIIF README](https://github.com/curationexperts/riiif/blob/master/README.md), or in the Hyrax installation instructions. 
+
+## 1. Ensure the RIIIF info service is running
+
+In your application somewhere you should have a block like the one below. In Hyrax 2.1+, it will be in `config/initializers/hyrax.rb`
+
+```ruby
+# Returns a URL that resolves to an info.json file provided by a IIIF image server
+config.iiif_info_url_builder = lambda do |file_id, base_url|
+  uri = Riiif::Engine.routes.url_helpers.info_url(file_id, host: base_url)
+  uri.sub(%r{/info\.json\Z}, '')
+end
+```
+
+Open your console and make sure that `iiif_info_url_builder` is returning something expected for a given `FileSet`. For example: 
+
+```ruby
+2.4.2 :014 > fs = FileSet.find("cz30ps64c")
+ => #<FileSet id: "cz30ps64c", head: [], tail: [], depositor: "bess@curationexperts.com", title: ["12219615_10153532796406998_4944957625326024288_n.jpg"], date_uploaded: "2018-05-06 14:32:42", date_modified: "2018-05-06 14:32:42", label: "12219615_10153532796406998_4944957625326024288_n.j...", relative_path: nil, import_url: nil, resource_type: [], creator: ["bess@curationexperts.com"], contributor: [], description: [], keyword: [], license: [], rights_statement: [], publisher: [], date_created: [], subject: [], language: [], identifier: [], based_near: [], related_url: [], bibliographic_citation: [], source: [], access_control_id: "c4cc4449-a032-4acf-b3b7-3b50a854f90e", embargo_id: nil, lease_id: nil>
+2.4.2 :015 > Riiif::Engine.routes.url_helpers.info_url(fs.files.first.id, host: "http://localhost:3000")
+ => "http://localhost:3000/images/cz30ps64c%2Ffiles%2F2ae0fa37-9052-460c-a56b-ab4603403f11/info.json"
+```
+
+Now you have a URL that should work to retrieve a IIIF manifest. Put it into a browser and see what that URL returns. It should give you something that looks like this:
```

Comment:
I think that "manifest" might be confusing here since I think "manifest" would normally refer to a IIIF Presentation json document.  It is a bit clunky but the Image API docs uses ["image information document"](http://iiif.io/api/image/2.1/#image-information-request-uri-syntax).